File: lksh/lksh2017ch/day0/F.py
n = int(input())
g = [[] for i in range(n)]
p = [int(i) for i in input().split()]

for i in range(1, n):
	g[p[i - 1] - 1].append(i)
	
ans = []
	
def dfs(v, pr, h):
	if (h % 2):
		ans.append(v + 1)
	for to in g[v]:
		if pr != to:
			dfs(to, v, h + 1)
	
dfs(0, 0, 0)

ans.sort()
print(len(ans))
print(*ans)

File: trains/urkop2009/F.cpp
#include <bits/stdc++.h>

using namespace std;

typedef long long ll;

int getSum(int x){
    int sum = 0;
    while(x){
        sum += x % 10;
        x /= 10;
    }
    return sum;
}

int main() {
#ifdef WIN
    freopen("01", "r", stdin);
#else
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
#endif // WIN
    ios_base::sync_with_stdio(0); cin.tie(0);
    int n;
    cin >> n;
    vector<pair<int, int> > t;
    for (int i = 1; i < 100000; i++) t.push_back(make_pair(getSum(i), i));
    sort(t.begin(), t.end());
    reverse(t.begin(), t.end());
    vector<int> ans;

    for (auto c : t){
        if (c.first <= n){
            n -= c.first;
            ans.push_back(c.second);
        }
    }

    cout << ans.size() << "\n";
    for (int x : ans) cout << x << ' ';
}

